pkg/collector/corechecks/oracle-dbm/statements.go

```diff
@@ -267,8 +296,32 @@ func (c *Check) StatementMetrics() (int, error) {
 
 	SQLTextErrors := 0
 	SQLCount := 0
+	totalSQLTextTimeUs := int64(0)
 	var oracleRows []OracleRow
-	if c.config.QueryMetrics {
+	FQTSent := make(map[string]int)
+	if c.config.QueryMetrics.Enabled {
+		if c.config.InstanceConfig.QueryMetrics.IncludeDatadogQueries {
+			var DDForceMatchingSignatures []string
+			err = c.db.Select(
+				&DDForceMatchingSignatures,
+				"SELECT distinct force_matching_signature FROM v$sqlstats WHERE sql_text like '%/* DD%' and (sysdate-last_active_time)*3600*24 < :1",
```

Comment:
is there a better way we can get these queries? like can we just filter for whatever user they configure for the ddagent? Or are you concerned that won't be distinct enough?

Reply:
> like can we just filter for whatever user they configure for the ddagent? Or are you concerned that won't be distinct enough?
[V$SQLSTATS](https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/V-SQLSTATS.html#GUID-495DD17D-6741-433F-871D-C965EB221DA9) doesn't contain the information about user, so we can filter on user. It doesn't contain any information about session related data either, so by we can't tell who was executing this query.

> is there a better way we can get these queries?
Filtering on the comment text seems straightforward to me. What are your concerns?

Reply:
It just feels a bit hacky, but if we can't look up the user then that seems fine

Reply:
The technical reason is that the SQL statements and their execution statistics are kept in cache, so the same statement can be run by many users and sessions. The accounting is done on the SQL cursor level.